test/functionalTest/cases/2015_notification_templates/notification_templates.test

```diff
@@ -0,0 +1,543 @@
+# Copyright 2016 Telefonica Investigacion y Desarrollo, S.A.U
+#
+# This file is part of Orion Context Broker.
+#
+# Orion Context Broker is free software: you can redistribute it and/or
+# modify it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# Orion Context Broker is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+# General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+#
+# For those usages not covered by this license please contact with
+# iot_support at tid dot es
+
+# VALGRIND_READY - to mark the test ready for valgrindTestSuite.sh
+
+--NAME--
+Notification Templates, Normal Case
+
+--SHELL-INIT--
+dbInit CB
+brokerStart CB
+accumulatorStart --pretty-print
+
+--SHELL--
+
+#
+# 01. Create subscription for E1 with only URL in template part
+# 02. Create subscription for E1 with URL and VERB in template part
+# 03. Create subscription for E1 with URL and PAYLOAD in template part - PAYLOAD with substitutions
+# 04. Create subscription for E1 with URL and QS in template part - QS with substitutions
+# 05. Create subscription for E1 with URL and HEADERS in template part - HEADERS with substitutions
+# 06. Create subscription for E1 with URL, VERB, PAYLOAD, QS and HEADERS in template part - all with substitutions (except VERB, of course)
+# 07. Create subscription for E1 with PAYLOAD, QS and HEADERS with unknown attribute 'A4' for substitution
+# 08. Create E1 with attributes A1,A2 and A3
+# 09. Dump accumulator, see 7 notifications
+#
+
```

Comment:
SORT_START / SORT_END  I had to add as the notifications don't always enter the accumulater in the same order. Not much I can do about that.
The `step` as Query String I used to be able to 'match' a notification with its subscription. As they don't always come in order, it was difficult without this 'mark'.